src/services/substrate-service.ts

```diff
@@ -129,6 +149,224 @@ export class SubstrateService {
+    const extrinsics = block.extrinsics.map((extrinsic) => {
+      const {
+        method,
+        nonce,
+        signature,
+        signer,
+        isSigned,
+        tip,
+        args,
+      } = extrinsic;
+      const convertedHash = u8aToHex(blake2AsU8a(extrinsic.toU8a(), 256));
+
+      return {
+        method: `${method.section}.${method.method}`,
+        signature: isSigned ? { signature, signer } : null,
+        nonce,
+        args,
+        // newArgs: this.parseGenericCall(method).args,
+        tip,
+        hash: convertedHash,
+        info: {},
+        events: [] as ISanitizedEvent[],
+        success: defaultSuccess,
+        // paysFee overrides to bool if `system.ExtrinsicSuccess|ExtrinsicFailed` event is present
+        paysFee: null as null | boolean,
+      };
+    });
+
+    const successEvent = "system.ExtrinsicSuccess";
+    const failureEvent = "system.ExtrinsicFailed";
+
+    if (Array.isArray(events)) {
+      for (const record of events) {
+        const { event, phase } = record;
+        const sanitizedEvent = {
+          method: `${event.section}.${event.method}`,
+          data: event.data,
+        } as any;
+
+        if (phase.isApplyExtrinsic) {
+          const extrinsicIdx = phase.asApplyExtrinsic.toNumber();
+          const extrinsic = extrinsics[extrinsicIdx];
+
+          if (!extrinsic) {
+            this.logger.error(
+              `Block ${block.header.number.toNumber()} ${blockHash}: Missing extrinsic ${extrinsicIdx}`
+            );
+            // eslint-disable-next-line no-continue
+            continue;
+          }
+
+          const method = `${event.section}.${event.method}`;
+
+          if (method === successEvent) {
+            extrinsic.success = true;
+          }
+
+          if (method === successEvent || method === failureEvent) {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const sanitizedData = event.data.toJSON() as any[];
+
+            for (const data of sanitizedData) {
+              // eslint-disablRegistrye-next-line @typescript-eslint/no-unsafe-member-access
+              if (data && data.paysFee) {
+                extrinsic.paysFee =
+                  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                  data.paysFee === true ||
+                  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                  data.paysFee === "Yes";
+
+                break;
+              }
+            }
+          }
+
+          extrinsic.events.push(sanitizedEvent);
+        } else if (phase.isFinalization) {
+          onFinalize.events.push(sanitizedEvent);
+        } else if (phase.isInitialization) {
+          onInitialize.events.push(sanitizedEvent);
+        }
+      }
+    }
+
+    // The genesis block is a special case with little information associated with it.
+    if (parentHash.every((byte) => !byte)) {
+      return {
```

Comment:
What is the diff between both these returns?

Reply:
Updated code with Polka store code.